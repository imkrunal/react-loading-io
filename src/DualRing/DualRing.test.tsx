import React from 'react'
import renderer from 'react-test-renderer'
import { DualRing } from './'

it('renders correctly', (): void => {
  const tree = renderer.create(<DualRing />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('joins default class with className prop', (): void => {
  const { root } = renderer.create(<DualRing className='spinner' />)

  expect(root.findByType('div').props.className).toEqual('rl-dual-ring spinner')
})

it('sets CSS vars with props', (): void => {
  const size = 100

  const { root } = renderer.create(
    <DualRing size={size} color='#d36ac2' speed={2} style={{ margin: 20 }} />
  )
  const { style } = root.findByType('div').props

  const scale = size / 200
  const translate = size * scale

  expect(style).toEqual({
    '--rl-dual-ring-size': '100px',
    '--rl-dual-ring-color': '#d36ac2',
    '--rl-dual-ring-scale': scale,
    '--rl-dual-ring-translate': `${translate}px`,
    '--rl-dual-ring-speed': `2s`,
    '--rl-dual-ring-border': `8px`,
    margin: 20
  })
})
