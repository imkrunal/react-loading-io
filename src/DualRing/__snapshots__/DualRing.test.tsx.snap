// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rl-dual-ring"
  style={
    Object {
      "--rl-dual-ring-border": "8px",
      "--rl-dual-ring-color": "#ED4585",
      "--rl-dual-ring-scale": 1,
      "--rl-dual-ring-size": "200px",
      "--rl-dual-ring-speed": "1s",
      "--rl-dual-ring-translate": "200px",
    }
  }
>
  <div />
</div>
`;
